tests/SbrpTests/ValidationTests.cs

```diff
@@ -0,0 +1,172 @@
+        }
+    }
+
+    private async Task CheckForSignatureAsync()
+    {
+        ISignatureVerificationProvider[] trustProviders = [new SignatureTrustAndValidityVerificationProvider()];
+        var verifier = new PackageSignatureVerifier(trustProviders);
+        var settings = SignedPackageVerifierSettings.GetDefault();
+
+        Output.WriteLine($"Checking {Packages.Count()} packages for signatures.");
+
+        foreach (var package in Packages)
+        {
+            bool isSigned = await IsPackageSignedAsync(package, verifier, settings);
+            Assert.False(isSigned, $"{package} is signed. Signed packages are not allowed in source-build-reference-packages.");
+        }
+    }
+
+    private bool HasSbrpAttribute(MetadataReader reader) =>
+        reader.CustomAttributes
+            .Select(attrHandle => reader.GetCustomAttribute(attrHandle))
+            .Any(attr => IsAttributeSbrp(reader, attr));
+
+    private bool IsAttributeSbrp(MetadataReader reader, CustomAttribute attr)
+    {
+        string attributeType = string.Empty;
+
+        if (attr.Constructor.Kind == HandleKind.MemberReference)
+        {
+            var mref = reader.GetMemberReference((MemberReferenceHandle)attr.Constructor);
+            if (mref.Parent.Kind == HandleKind.TypeReference)
+            {
+                var tref = reader.GetTypeReference((TypeReferenceHandle)mref.Parent);
+                attributeType = $"{reader.GetString(tref.Namespace)}.{reader.GetString(tref.Name)}";
+            }
+            else if (mref.Parent.Kind == HandleKind.TypeDefinition)
+            {
+                var tdef = reader.GetTypeDefinition((TypeDefinitionHandle)mref.Parent);
+                attributeType = $"{reader.GetString(tdef.Namespace)}.{reader.GetString(tdef.Name)}";
+            }
+        }
+        else if (attr.Constructor.Kind == HandleKind.MethodDefinition)
+        {
+            var mdef = reader.GetMethodDefinition((MethodDefinitionHandle)attr.Constructor);
+            var tdef = reader.GetTypeDefinition(mdef.GetDeclaringType());
+            attributeType = $"{reader.GetString(tdef.Namespace)}.{reader.GetString(tdef.Name)}";
+        }
+
+        if (attributeType == SbrpAttributeType)
+        {
+            var decodedValue = attr.DecodeValue(DummyAttributeTypeProvider.Instance);
+            try
+            {
+                return decodedValue.FixedArguments[0].Value?.ToString() == "source" && decodedValue.FixedArguments[1].Value?.ToString() == "source-build-reference-packages";
+            }
+            catch
+            {
+                throw new InvalidOperationException($"{SbrpAttributeType} is not formatted properly with a key, value pair.");
+            }
+        }
+
+        return false;
+    }
+
+    public async Task<bool> IsPackageSignedAsync(string packagePath, PackageSignatureVerifier verifier, SignedPackageVerifierSettings settings)
+    {
+        using var packageReader = new PackageArchiveReader(packagePath);
```

Comment:
Consider making use of the language feature to simplify by removing the var usage.  This comment applies in several places.

```suggestion
        using PackageArchiveReader packageReader = new (packagePath);
```

Reply:
Fixed in https://github.com/dotnet/source-build-reference-packages/pull/817/commits/2aae3ceba8c63acabf1af5acaeded0330785d706